libr/core/cmd_search_rop.c

```diff
@@ -52,9 +52,101 @@ static RList* get_constants (const char *str) {
 	return list;
 }
 
+static bool isFlag (const char *f) {
+	const char *flags = "zf,pf,sf,cf,of";
+
+	if (strstr (flags, f)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+// TODO: hideous code, must rafactor
+// binary op
+static ut64 simulate_op (const char *op, ut64 src1, ut64 src2, ut64 old_src1, ut64 old_src2) {
+
+	if (!strcmp (op, "^")) {
+		return src1 ^ src2;
+	} else if (!strcmp (op, "+")) {
+		return src1 + src2;
+	} else if (!strcmp (op, "-")) {
+		return src1 - src2;
+	} else if (!strcmp (op, "*")) {
+		return src1 * src2;
+	} else if (!strcmp (op, "/")) {
+		return src1 / src2;
+	} else if (!strcmp (op, "%")) {
+		return src1 % src2;
+	} else if (!strcmp (op, "<<")) {
+		return src1 << src2;
+	} else if (!strcmp (op, ">>")) {
+		return src1 >> src2;
+	} else if (!strcmp (op, "&")) {
+		return src1 & src2;
+	} else if (!strcmp (op, "+=")) {
+		return old_src1 + src2;
+	} else if (!strcmp (op, "-=")) {
+		return old_src1 - src2;
+	} else if (!strcmp (op, "*=")) {
+		return old_src1 * src2;
+	} else if (!strcmp (op, "/=")) {
+		return old_src1 / src2;
+	} else if (!strcmp (op, "%=")) {
+		return old_src1 % src2;
+	} else if (!strcmp (op, "<<")) {
+		return src1 << src2;
+	} else if (!strcmp (op, ">>")) {
+		return src1 >> src2;
+	} else if (!strcmp (op, "&=")) {
+		return src1 & src2;
+	} else if (!strcmp (op, "^=")) {
+		return src1 ^ src2;
+	} else if (!strcmp (op, "|=")) {
+		return src1 | src2;
+	}
+
+	// fail value
+	return -1337;
+}
+
+// fill REGs with rwx addr from MEM
+static void fillRegisterValues (RCore *core) {
+	RIOSection *sec;
+	RListIter *iter_io, *iter_reg;
+	RList *regs;
+	RRegItem *reg_item;
+	ut64 addr;
+
+	regs = r_reg_get_list (core->dbg->reg, 0);
+	if (!regs) {
+		return;
+	}
+
+	iter_reg = regs->head;
+	reg_item = iter_reg->data;
+	r_reg_arena_pop (core->dbg->reg);
+	r_list_foreach (core->io->sections, iter_io, sec) {
+		if (sec->rwx & 1) {
+			addr = sec->vaddr;
+			// r_cons_printf ("RWX addr = 0x%"PFMT64x"\n", addr);
+			if (iter_reg && reg_item) {
```

Comment:
cant we just move this check outside the loop, right after setting those vars? and drop the return statement here